src/cli/Logger.js

```
@@ -67,23 +101,94 @@ class Logger {
     this.verboseMode = true;
     // Flush all previous verbose logs to the output
     this.verboseLogs.forEach(({ message, namespace }) => {
-      this.verbose(message, namespace);
+      this.doLogVerbose(message, namespace);
     });
     this.verboseLogs = [];
   }
 
   /**
    * @private
-   * @param {string[]} namespace
+   * @param {string} message
+   * @param {string[]} [namespace]
+   */
+  doLogVerbose(message, namespace) {
+    message = colors.gray(message);
+    message = this.namespaceLogMessage(message, namespace);
+    this.safeWrite(message, false);
+  }
+
+  /**
+   * @private
+   * @param {string} text
+   */
+  writeToLogsFile(text) {
+    text = stripAnsi(text);
+    this.logsFileStream.write(`${text}\n`);
+  }
+
+  /**
+   * @private
+   * @param {string|undefined} text
+   * @param {string[]} [namespace]
+   * @return {string}
+   */
+  namespaceLogMessage(text, namespace) {
+    text = text ?? '';
+    const prefix = this.generatePrefix(namespace);
+    return text
+      .split('\n')
+      .map((line) => `${prefix}${line}`)
+      .join('\n');
+  }
+
+  /**
+   * @private
+   * @param {string[]} [namespace]
    * @return string
    */
   generatePrefix(namespace) {
+    if (!namespace) {
+      return '';
+    }
     let prefix = namespace.join(` ${symbols.separator} `);
     if (prefix.length > 0) {
       prefix = colors.gray(`${prefix} ${symbols.separator} `);
     }
     return prefix;
   }
+
+  openLogsFile() {
+    const directory = path.dirname(this.logsFilePath);
+    if (!fs.existsSync(directory)) {
+      fs.mkdirSync(directory);
+    }
+    return fs.createWriteStream(this.logsFilePath);
```

What if, for some reason, we cannot create/open the logs file? I think we should handle such situation gracefully instead of crashing (which I'm assuming would be the case at the moment)

---

Or did you mean rather to throw a clearer exception?